src/Microsoft.DotNet.CodeFormatting/NameHelper.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.CodeFormatting
+{
+    /// <summary>
+    /// Commonly used name functions
+    /// </summary>
+    internal static class NameHelper
+    {
+        /// <summary>
+        /// Get the full name of a symbol. So a field might look like
+        /// "OuterNamespace.Inner.ClassName.FieldName"
+        /// </summary>
+        /// <param name="symbol">symbol to get name of</param>
+        /// <returns>Full display name for a symbol</returns>
+        internal static string GetFullName(ISymbol symbol)
+        {
+            return GetFullName(symbol.ContainingType) + "." + symbol.Name;
```

Comment:
Indeed so. Even though it’s only sort of a helper function internally (these names are only ever used to track down a type for a check), I’m going to update it to use the +, since it’s correct.

From: Jared Parsons [mailto:notifications@github.com]
Sent: Monday, April 13, 2015 2:54 PM
To: dotnet/codeformatter
Cc: Chad Nedzlek
Subject: Re: [codeformatter] Readonly field formatter (#109)


In src/Microsoft.DotNet.CodeFormatting/NameHelper.cs<https://github.com/dotnet/codeformatter/pull/109#discussion_r28283251>:

> +namespace Microsoft.DotNet.CodeFormatting

> +{

> +    /// <summary>

> +    /// Commonly used name functions

> +    /// </summary>

> +    internal static class NameHelper

> +    {

> +        /// <summary>

> +        /// Get the full name of a symbol. So a field might look like

> +        /// "OuterNamespace.Inner.ClassName.FieldName"

> +        /// </summary>

> +        /// <param name="symbol">symbol to get name of</param>

> +        /// <returns>Full display name for a symbol</returns>

> +        internal static string GetFullName(ISymbol symbol)

> +        {

> +            return GetFullName(symbol.ContainingType) + "." + symbol.Name;

Don't believe this will work for nested types. They should use + instead of .

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/codeformatter/pull/109/files#r28283251>.